lib/bacnet-asn1.js

```diff
@@ -779,7 +784,14 @@ var decode_read_access_result = module.exports.decode_read_access_result = funct
         var localResult = bacapp_decode_application_data(buffer, offset + len, apduLen + offset - 1, value.objectIdentifier.type, new_entry.propertyIdentifier);
         if (!localResult) return;
         len += localResult.len;
-        localValues.push({value: localResult.value, type: localResult.type});
+        var resObj = {
+          value: localResult.value,
+          type: localResult.type
+        };
+        if (localResult.encoding !== undefined) {
```

Comment:
Absolutely right, but why do the higher level functions even know about `value` and `type`? Couldn't we just pass through what we receive? And hey, shouldn't it be `tag` instead of `type`? I think there is also a little confusion when to use which of those two (object vs. property).